src/Toolkit/Toolkit/UI/Controls/PopupViewer/PopupMediaView.cs

```diff
@@ -214,11 +216,17 @@ public PopupMedia? PopupMedia
 
         private void OnPopupMediaPropertyChanged()
         {
-            // TODO: Handle PopupMedia.ImageRefreshInterval - Start/Stop/Reset refresh timer if "interval > 0 && IsLoaded==true" + start/stop on load/unload
+            // Stop any existing timer
+            StopRefreshTimer();
             _lastChartSize = 0;
             if (PopupMedia?.Type == PopupMediaType.Image)
             {
                 UpdateImage();
+                // Start the refresh timer if the interval is greater than zero
+                if (PopupMedia.ImageRefreshInterval > TimeSpan.Zero && IsLoaded)
+                {
+                    StartRefreshTimer(PopupMedia.ImageRefreshInterval);
```

Comment:
The timer should not run if the view isn't loaded. It should also stop on unload.